.github/workflows/beta.yml

```diff
@@ -0,0 +1,80 @@
+name: Beta release
+
+run-name: "Beta release '${{ inputs.git-ref || github.ref_name }}' (publish: ${{ inputs.publish || github.event_name == 'schedule' }})"
+
+on:
+  workflow_dispatch:
+    inputs:
+      git-ref:
+        required: true
+        type: string
+        description: "The github ref of this beta version (i.e. main, 1234567)"
+        default: main
+      publish:
+        required: false
+        type: boolean
+        default: false
+        description: "Publish the beta release"
+  schedule:
+    - cron: '0 0 * * TUE'
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  prepare-vars:
+    name: Prepare vars
+    if: ${{ inputs.publish || github.event_name == 'schedule' }}
+    runs-on: ubuntu-latest
+    outputs:
+      num-tuesday: ${{ steps.outputs.outputs.git-ref }}
+    steps:
+      - name: Set outputs
+        id: outputs
+        run: |
+          # Find out the number of Tuesday in a month this is
+          echo "num-tuesday=$((($(date +%d)-1)/7+1))" >> $GITHUB_OUTPUT
+      
+  crate:
+    name: Publish surrealdb to crates.io
+    needs: [prepare-vars]
+    # We are only going to release the beta version every second or fourth Tuesday
+    if: ${{ needs.prepare-vars.outputs.num-tuesday == 2 || needs.prepare-vars.outputs.num-tuesday == 4 }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install stable toolchain
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: stable
+
+      - name: Checkout sources
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.git-ref || github.ref_name }}
+
+      - name: Publish
+        env:
+          CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
+        run: |
+          # Install release-plz
+          cargo install --force --locked --version 0.3.30 release-plz
+
+          # Bump the version
+          /home/runner/.cargo/bin/release-plz update --config .config/release-plz.toml
+
+          # And append "-beta.1" at the end
+          sed -i "s#^version = \"\(.*\)\"#version = \"\1-beta.1\"#" lib/Cargo.toml
```

Comment:
should this be `-beta.1` instead of `1-beta.1`?

Reply:
also, do we expect to have `beta.2`? how is that handled?

Reply:
> should this be -beta.1 instead of 1-beta.1?

It's not `1-beta.1`. The `\1` is the result of the regex match. We are simply getting the new (bumped) version from `Cargo.toml` and then adding `-beta.1` to it. In this case, it will end up as `1.1.0-beta.1`.

Reply:
> also, do we expect to have beta.2? how is that handled?

This particular workflow is meant for the initial beta only for now. We will either extend it to support subsequent betas or create a new workflow.

Reply:
got it 👍